test/stdlib/Compression.swift

```diff
@@ -0,0 +1,137 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+// REQUIRES: objc_interop
+
+import StdlibUnittest
+import Compression
+import Foundation
+
+// Read from Data
+class DataSource {
+
+  private var _buf : Data
+  private var _bufSize : Int // total byte count
+  private var _pos : Int  // next byte to read
+
+  public init(_ d: Data) {
+    _buf = d
+    _bufSize = d.count
+    _pos = 0
+  }
+
+  public func readData(ofLength len: Int) -> Data? {
+    let n = min(len,_bufSize - _pos)
+    if (n == 0) { return nil } // EOF
+    let d = _buf.subdata(in: _pos ..< _pos + n)
+    _pos += n
+    return d
+  }
+
+}
+
+@available(macOS 10.12, iOS 10.0, watchOS 3.0, tvOS 10.0, *)
+func ofiltercompress_ifilterdecompress(_ contents: Data, using algo: Algorithm) throws -> Bool {
+
+  var payload = Data()
+  var decompressed = Data()
+
+  // Output filter compress
+  let f = DataSource(contents)
+  let ofilter = try Compression.OutputFilter(.compress, using: algo) { (x:Data?) -> () in if x != nil { payload.append(x!) } }
+  while (true) {
+    let i = f.readData(ofLength: 900)
+    try ofilter.write(i) // will finalize when i is empty
+    if i == nil { break }
+  }
+
+  // Input filter decompress
+  let g = DataSource(payload)
+  let ifilter = try InputFilter(.decompress, using: algo, readingFrom: { (x:Int) -> Data? in return g.readData(ofLength:x) } )
+  while (true) {
+    let i = try ifilter.readData(ofLength: 400)
+    if i == nil { break }
+    decompressed.append(i!)
+  }
+
+  print("ofilter | ifilter \(algo): \(contents.count) -> \(payload.count) -> \(decompressed.count)")
+  let pass = (contents == decompressed)
+  if !pass { print("FAIL") }
+
+  return pass
+}
+
+@available(macOS 10.12, iOS 10.0, watchOS 3.0, tvOS 10.0, *)
+func ifiltercompress_ofilterdecompress(_ contents: Data, using algo: Algorithm) throws -> Bool {
+
+  var payload = Data()
+  var decompressed = Data()
+
+  // Input filter compress
+  let f = DataSource(contents)
+  let ifilter = try InputFilter(.compress, using: algo, readingFrom: { (x:Int) -> Data? in return f.readData(ofLength:x) } )
+  while (true) {
+    let i = try ifilter.readData(ofLength: 400)
+    if i == nil { break }
+    payload.append(i!)
+  }
+
+  // Output filter decompress
+  let g = DataSource(payload)
+  let ofilter = try OutputFilter(.decompress, using: algo) { (x:Data?) -> () in if x != nil { decompressed.append(x!) } }
+  while (true) {
+    let i = g.readData(ofLength: 900)
+    try ofilter.write(i) // will finalize when i is empty
+    if i == nil { break }
+  }
+
+  print("ifilter | ofilter \(algo): \(contents.count) -> \(payload.count) -> \(decompressed.count)")
+  let pass = (contents == decompressed)
+  if !pass { print("FAIL") }
+
+  return pass
+}
+
+@available(macOS 10.12, iOS 10.0, watchOS 3.0, tvOS 10.0, *)
+func test_all() throws -> Int { // return number of failed cases
```

Comment:
It would be greatly appreciated if you refactored these into `tests.test` calls. This way any failures will be way easier to debug and fix. Let me know if you have capacity to do that or need help.